memcache/selector.go

```diff
@@ -67,25 +64,30 @@ func (s *staticAddr) String() string  { return s.str }
 // is returned, no changes are made to the ServerList.
 func (ss *ServerList) SetServers(servers ...string) error {
 	naddr := make([]net.Addr, len(servers))
+	strAddrs := make(map[string]net.Addr)
 	for i, server := range servers {
 		if strings.Contains(server, "/") {
 			addr, err := net.ResolveUnixAddr("unix", server)
 			if err != nil {
 				return err
 			}
+			strAddrs[server] = newStaticAddr(addr)
 			naddr[i] = newStaticAddr(addr)
 		} else {
 			tcpaddr, err := net.ResolveTCPAddr("tcp", server)
 			if err != nil {
 				return err
 			}
 			naddr[i] = newStaticAddr(tcpaddr)
+			strAddrs[server] = newStaticAddr(tcpaddr)
 		}
 	}
 
 	ss.mu.Lock()
 	defer ss.mu.Unlock()
 	ss.addrs = naddr
+	ss.strAddrs = strAddrs
```

Comment:
Can we remove ss.addrs if not required. As I can see the ss.strAddrs has the same data init.